cirq/work/observable_measurement_data.py

```diff
@@ -0,0 +1,478 @@
+            f'chunksizes={proper_repr(self.chunksizes)}, '
+            f'timestamps={proper_repr(self.timestamps)})'
+        )
+
+    def __str__(self):
+        s = f'Accumulator {self.max_setting}; {self.n_repetitions} repetitions\n'
+        s += '\n'.join('  ' + self.summary_string(setting) for setting in self._simul_settings)
+        return s
+
+    def covariance(self, *, atol=1e-8) -> np.ndarray:
+        """Compute the covariance matrix for the estimators of all settings.
+
+        Like `variance`, this is the covariance of the sampling distribution
+        of the sample mean. Practically, it is the 'normal' covariance
+        divided by the number of observations (bitstrings).
+
+        Args:
+            atol: The absolute tolerance for asserting coefficients are real.
+        """
+        if len(self.bitstrings) == 0:
+            # coverage: ignore
+            raise ValueError("No measurements")
+
+        all_obs_vals = np.array(
+            [
+                _obs_vals_from_measurements(
+                    bitstrings=self.bitstrings,
+                    qubit_to_index=self._qubit_to_index,
+                    observable=setting.observable,
+                    atol=atol,
+                )
+                for setting in self._simul_settings
+            ]
+        )
+
+        # https://github.com/numpy/numpy/issues/11502
+        if all_obs_vals.shape[0] == 1:
+            # coverage: ignore
+            cov = np.array([[np.var(all_obs_vals[0], ddof=1)]])
+            return cov
+
+        cov = np.cov(all_obs_vals, ddof=1) / all_obs_vals.shape[1]
+        return cov
+
+    def _validate_setting(self, setting: InitObsSetting, what: str):
+        mws = _max_weight_state([self.max_setting.init_state, setting.init_state])
+        mwo = _max_weight_observable([self.max_setting.observable, setting.observable])
+        if mws is None or mwo is None:
+            raise ValueError(
+                f"You requested the {what} for a setting that is not compatible "
+                f"with this BitstringAccumulator's meas_spec."
+            )
+
+    def variance(self, setting: InitObsSetting, *, atol: float = 1e-8):
+        """Compute the variance of the estimators of the given setting.
+
+        This is the normal variance divided by the number of samples to estimate
+        the certainty of our estimate of the mean. It is the standard error
+        of the mean, squared.
+
+        This uses `ddof=1` during the call to `np.var` for an unbiased estimator
+        of the variance in a hypothetical infinite population for consistency
+        with `BitstringAccumulator.covariance()` but differs from the default
+        for `np.var`.
+
+        Args:
+            setting: The setting
+            atol: The absolute tolerance for asserting coefficients are real.
+        """
+        if len(self.bitstrings) == 0:
+            # coverage: ignore
+            raise ValueError("No measurements")
```

Comment:
Do you mind writing the test for these errors?

Reply:
done